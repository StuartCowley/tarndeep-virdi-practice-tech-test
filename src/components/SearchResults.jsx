import React from "react";
import "../styles/SearchResults.css";

export default function SearchResults({ results }) {
  let display = results.map((result) => {
    return <img className="result--image" src={result} alt="" />;
  });

  return (
    <>
      {results && results.length === 0 ? (
        <p>No Results</p>
      ) : (
        <div className="result--table">{display}</div>
      )}
    </>
  );
}
