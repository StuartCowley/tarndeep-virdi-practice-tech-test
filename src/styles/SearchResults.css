.result--table {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  grid-gap: 10px;
}

.result--image {
  width: 100%;
  height: auto;
  object-fit: cover;
}

@media (max-width: 769px) {
  .result--table {
    grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
  }
}
