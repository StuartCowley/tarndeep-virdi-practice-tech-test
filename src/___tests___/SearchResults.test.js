import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults component", () => {
  const validProps = [
    ["test-url-1", "alt-1"],
    ["test-url-2", "alt-2"],
    ["test-url-3", "alt-3"],
    ["test-url-4", "alt-4"],
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults results={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders image correctly", () => {
    const result = [
      [
        "https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80",
        "Image of the Moon",
      ],
    ];
    render(<SearchResults results={result} />);

    expect(screen.getByAltText("Image of the Moon")).toBeInTheDocument();
  });

  it("renders 'No Results' message when there are no search results", () => {
    render(<SearchResults results={[]} />);

    expect(screen.getByText("No Results")).toBeInTheDocument();
  });
});
