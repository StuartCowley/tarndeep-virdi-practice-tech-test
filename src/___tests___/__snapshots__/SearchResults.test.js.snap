// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results__image-table"
  >
    <img
      alt="alt-1"
      class="search-results__image"
      src="test-url-1"
    />
    <img
      alt="alt-2"
      class="search-results__image"
      src="test-url-2"
    />
    <img
      alt="alt-3"
      class="search-results__image"
      src="test-url-3"
    />
    <img
      alt="alt-4"
      class="search-results__image"
      src="test-url-4"
    />
  </div>
</DocumentFragment>
`;
